# Download the kaggle data set.

In [0]:

!pip install -q kaggle
!mkdir -p ~/.kaggle
!cp kaggle.json ~/.kaggle
!ls ~/.kaggle
!chmod 600 /root/.kaggle/kaggle.json

!kaggle competitions download -c humpback-whale-identification
!rm -rf input
!mkdir -p input
!unzip -q train.zip -d input/train
#!ls input/train

kaggle.json
  0% 0.00/498k [00:00<?, ?B/s]
100% 498k/498k [00:00<00:00, 74.1MB/s]
  0% 0.00/594k [00:00<?, ?B/s]
100% 594k/594k [00:00<00:00, 80.4MB/s]
100% 1.34G/1.35G [00:08<00:00, 148MB/s]
100% 1.35G/1.35G [00:08<00:00, 169MB/s]
100% 4.14G/4.16G [00:26<00:00, 199MB/s]
100% 4.16G/4.16G [00:26<00:00, 166MB/s]


In [0]:
import numpy as np 
import pandas as pd 
import os
import matplotlib.pyplot as plt
import matplotlib.image as mplimg
from matplotlib.pyplot import imshow

from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import OneHotEncoder

from sklearn.model_selection import train_test_split


from keras import layers
from keras.preprocessing import image
from keras.applications.imagenet_utils import preprocess_input
from keras.layers import Input, Dense, Activation, BatchNormalization, Flatten, Conv2D
from keras.layers import AveragePooling2D, MaxPooling2D, Dropout
from keras.models import Model

import keras.backend as K
from keras.models import Sequential

import warnings
warnings.simplefilter("ignore", category=DeprecationWarning)


Using TensorFlow backend.


# Explore the training data.
Let us explore the training data.


In [0]:
train_df = pd.read_csv("train.csv")
train_df.head()

Image         Id
0  0000e88ab.jpg  w_f48451c
1  0001f9222.jpg  w_c3d896a
2  00029d126.jpg  w_20df2c5
3  00050a15a.jpg  new_whale
4  0005c1ef8.jpg  new_whale

In [0]:
train_df.Id.value_counts()

new_whale    9664
w_23a388d      73
w_9b5109b      65
w_9c506f6      62
w_0369a5c      61
w_700ebb4      57
w_3de579a      54
w_564a34b      51
w_fd3e556      50
w_88e4537      49
w_2b069ba      48
w_d405854      47
w_789c969      45
w_f0fe284      45
w_778e474      40
w_5e8e218      40
w_343f088      40
w_a9304b9      37
w_60ce6fc      37
w_5a2634c      37
w_6822dbc      36
w_af367c3      35
w_f765256      34
w_1ca9ab1      34
w_17b0d3a      33
w_d72771c      32
w_08630fd      31
w_8c25681      31
w_6cda039      31
w_fd6a3ad      30
             ... 
w_55782b0       1
w_a4a0867       1
w_68d1fb0       1
w_2856c57       1
w_81d807a       1
w_05f8c99       1
w_8d0d921       1
w_6441130       1
w_ea40c36       1
w_a33d1c0       1
w_d869993       1
w_1dd336a       1
w_ff3efba       1
w_1728040       1
w_4df6c4b       1
w_3514a84       1
w_b60f441       1
w_b919aad       1
w_ea67beb       1
w_19a0b30       1
w_6d527ff       1
w_c67271f       1
w_d12e2eb       1
w_fbe3f59       1
w_db88b61 

In [0]:
print('Number of rows in train.csv', len(train_df))

Number of rows in train.csv 25361


# Identify the data points

## train.csv
There are 25361 rows in train.csv.  Which corresponds to the image entries in train.zip
We can see that the train.csv file has two data fields.  
* Image : The whale image file name
* Id is the whale Id.
Each whale is assigned a unique Id.  The unidentified whale's are assigned an Id new_whale.  


## train.zip
There are 25361 image files in train.zip file.  It has been extracted to input/train folder.  The filename corresponds to the Image column in train.csv file.

# Split the data into training, validation & test datasets





In [0]:
x = train_df.Image
y = train_df.Id

X_train, X_tmp, Y_train, Y_tmp = train_test_split(x, y, train_size=0.8, random_state=1)

X_val, X_test, Y_val, Y_test   = train_test_split(X_tmp, Y_tmp, train_size=0.5, random_state=1)

print('Training, Validation & testing data size', len(X_train),len(X_val), len(X_test))


Training, Validation & testing data size 20288 2536 2537
